### ISSUE TITLE
Remove redundant check list in `quickFix.ts` and `FixableCheck`

### ISSUE THREAD

**Issue by karlvr** [jdneo/vscode-checkstyle#299]
It's a trifle annoying to have to add checks to `quickFix.ts` in additional to `FixableCheck`. What do you think about having the extension load the list of checks from the Java component, so it doesn't need to maintain its own list, but it can still decide whether a `Diagnostic` is fixable without invoking the Java code.

Also `FixableCheck` could be removed if we instead put the check names into the quick fix classes themselves, and then `QuickFixServer` could build a list of the supported checks and the matching quick fixes.

e.g.

```java

abstract public class BaseQuickFix {

    public abstract String[] getSupportedChecks();

}
```

**Comment by jdneo** [comment]
I agree that distributing such quickfix information everywhere is a bad way for implementation. And sometime it introduces bugs. For example, when we have fix for the violation but for some reason the fix failed to generate the worksapce edit, then the UX goes into an unresponsive state.

I would be great if the quickfix list can be rendered on demand!
